docs/Execution.md

```diff
@@ -0,0 +1,107 @@
+Execution
+======
+
+The MsQuic API uses a different execution model compared to BSD-style sockets and most other networking libraries built on them.
+The sections below detail the designs MsQuic uses and the reasons behind these choices.
+
+## Event Model
+
+In the MsQuic API, all state changes and notifications are indicated directly to the application via a callback.
```

Comment:
all "asynchronous" state changes? presumably some APIs are synchronous, so QuicBind() could return QUIC_SUCCESS and implicitly transition state?

Reply:
I cannot think of any blocking APIs beyond getting/setting parameters (i.e. socket options). Everything is async. Your notion of a 'bind' is simply setting a 'local address' parameter before calling the `ConnectionStart` async function, which then will attempt to use that local address.

Reply:
ok, maybe i'm splitting hairs here, but i still suggest that "asynchronous" be added here, because calling `ConnectionStart` implicitly changes the state of a connection from "doesn't exist" or "not started" etc., to "starting" from the perspective of an application. just because an explicit notification wasn't created for that state change doesn't make it less of a conceptual state change, imo.

Reply:
Added.